timer.js

```diff
@@ -11,10 +11,32 @@
     // Like setTimeout, but only takes a function argument.  There's
     // no time argument (always zero) and no arguments (you have to
     // use a closure).
-    function setZeroTimeout(fn) {
+    function nextTickBeforeEvents(fn) {
         resolved.then(fn);
     }
 
     // Add the one thing we want added to the window object.
-    window.setZeroTimeout = setZeroTimeout;
+    window.nextTickBeforeEvents = nextTickBeforeEvents;
+})();
+
+(function() {
+  var cbs = [];
+  var msg = 135921;
+
+  function nextTickDuringEvents(fn) {
+    cbs.push(fn);
+    window.postMessage(msg, window.location.origin);
+  }
+
+  function recv(ev) {
+    if (window !== ev.source || ev.data !== msg) {
+      return;
+    }
+
+    cbs.shift()();
+  }
+
+  window.addEventListener("message", recv, false);
```

Comment:
The [original implementation](https://github.com/mozilla/pluotsorbet/commit/f351476bfb4ccc7466faa611d06a4c3f0e916f44), which appears to have been based on [this dbaron blog post](http://dbaron.org/log/20100309-faster-timeouts), has three differences:

1. The target origin is the wildcard "*".
2. The event listener is capturing.
3. The event listener calls *stopPropagation* on the event.

I'm unsure if any of these are substantive. The capturing and *stopPropagation* call might be to prevent third-party code from listening in, which shouldn't matter in our case. Did you make any of these changes on purpose?